docs/integrations/pydantic.md

```diff
@@ -313,3 +313,119 @@ class Query:
+unrepresentable in GraphQL (such as unions of scalar values) or structural
+changes are needed before the data is exposed in the schema. In these cases,
+there are two methods you can use to control the conversion logic more directly.
+
+First, you can use a different type annotation in your Strawberry model for a
+field type instead of using `strawberry.auto` to choose an equivalent type. This
+allows you to do things like converting values to custom scalar types or
+converting between basic types. Strawberry will call the constructor of the new
+type annotation with the field value as input, so this only works when
+conversion is possible through a constructor.
+
+```python
+import base64
+import strawberry
+from pydantic import BaseModel
+from typing import Union, NewType
+
+
+class User(BaseModel):
+    id: Union[int, str]  # Not representable in GraphQL
+    hash: bytes
+
+
+Base64 = strawberry.scalar(
+    NewType("Base64", bytes),
+    serialize=lambda v: base64.b64encode(v).decode("utf-8"),
+    parse_value=lambda v: base64.b64decode(v.encode("utf-8")),
+)
+
+
+@strawberry.experimental.pydantic.type(model=User)
+class UserType:
+    id: str  # Serialize int values to strings
+    hash: Base64  # Use a custom scalar to serialize values
+
+
+@strawberry.type
+class Query:
+    @strawberry.field
+    def test() -> UserType:
+        return UserType.from_pydantic(User(id=123, hash=b'abcd'))
+
+
+schema = strawberry.Schema(query=Query)
+
+print(schema.execute_sync("query { test { id, hash } }").data)
+# {"test": {"id": "123", "hash": "YWJjZA=="}}
+```
+
+The other, more comprehensive, method for modifying the conversion logic is to
+provide custom implementations of `from_pydantic` and `to_pydantic`. This allows
+you full control over the conversion process and bypasses Strawberry's built in
+conversion rules completely, while still registering the new type as a Pydantic
+conversion type so it can be referenced in other models.
+
+This is useful when you need to represent structures that are very different
+from GraphQL standards, without changing the underlying Pydantic model. An
+example would be a use case that uses a `dict` field to store some
+semi-structured content, which is difficult to represent in GraphQL's strict
+type system.
+
+```python
+import enum
+import dataclasses
+import strawberry
+from pydantic import BaseModel
+from typing import Any, Dict, Optional
+
+
+class ContentType(enum.Enum):
+    NAME = "name"
+    DESCRIPTION = "description"
+
+class User(BaseModel):
+    id: str
+    content: Dict[ContentType, str]
+
+@strawberry.experimental.pydantic.type(model=User)
+class UserType:
+    id: strawberry.auto
+    # Flatten the content dict into specific fields in the query
+    content_name: Optional[str] = None
+    content_description: Optional[str] = None
+
+    @staticmethod
+    def from_pydantic(user: User, extra: Dict[str, Any] = None) -> "UserType":
```

Comment:
i'm not sure whether we should enforce / encourage that the signature should have the keywords "instance", and "extra" 
as specified in the protocol `StrawberryTypeFromPydantic`. 

```suggestion
    def from_pydantic(instance: User, extra: Dict[str, Any] = None) -> "UserType":
```

currently conversion logic wise, it shouldn't affect anything since no code calls from_pydantic. 
but both mypy and pycharm will think that the keyword is "instance" instead of "user" here. 
so people may pass in the wrong keyword.

Reply:
<img width="935" alt="image" src="https://user-images.githubusercontent.com/30519287/152276909-4a3c56f5-362e-4ac5-bfec-f6bd2d6c9ce4.png">

Reply:
Fixed the example at least, I'm not sure if we should enforce it though since using another name does technically work aside from IDEs.